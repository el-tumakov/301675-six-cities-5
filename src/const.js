export const FEATURES = [
  `Wi-Fi`,
  `Washing machine`,
  `Towels`,
  `Heating`,
  `Coffee machine`,
  `Baby seat`,
  `Kitchen`,
  `Dishwasher`,
  `Cabel TV`,
  `Fridge`
];

export const OFFERS_TYPE = [
  `apartment`,
  `room`,
  `house`,
  `hotel`
];
