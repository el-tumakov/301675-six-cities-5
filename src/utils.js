export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const adaptOffersToClient = (offers) => {
  const adaptedOffers = offers.map((offer) => {
    const adaptedOffer = Object.assign(
        {},
        offer,
        {
          owner: {
            id: offer.host.id,
            avatar: offer.host.avatar_url,
            name: offer.host.name,
            super: offer.host.is_pro
          },
          photos: offer.images,
          favorite: offer.is_favorite,
          premium: offer.is_premium,
          guests: offer.max_adults,
          preview: offer.preview_image,
          features: offer.goods
        }
    );

    delete adaptedOffer.host;
    delete adaptedOffer.images;
    delete adaptedOffer.is_favorite;
    delete adaptedOffer.is_premium;
    delete adaptedOffer.max_adults;
    delete adaptedOffer.preview_image;
    delete adaptedOffer.goods;


    return adaptedOffer;
  });

  return adaptedOffers;
};

export const adaptReviewsToClient = (reviews) => {
  const adaptedReviews = reviews.map((review) => {
    const adaptedReview = Object.assign(
        {},
        review,
        {
          user: Object.assign({}, review.user, {
            avatar: review.user.avatar_url,
            isPro: review.user.is_pro
          })
        }
    );

    delete adaptedReview.user.avatar_url;
    delete adaptedReview.user.is_pro;

    return adaptedReview;
  });

  return adaptedReviews;
};
