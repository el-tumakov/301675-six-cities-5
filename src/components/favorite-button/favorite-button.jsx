import React, {useState} from "react";
import PropTypes from "prop-types";
import {useHistory} from "react-router-dom";
import {connect} from "react-redux";
import {setFavorite} from "../../store/api-actions";
import {AuthorizationStatus} from "../../const";


const FavoriteButton = (props) => {
  const {id, favorite, authorizationStatus, onButtonClick} = props;
  const history = useHistory();
  const [isFavorite, changeFavorite] = useState(favorite);

  return (
    <button
      className={
        `${
          isFavorite
            ? `place-card__bookmark-button--active`
            : ``
        } place-card__bookmark-button button`
      }
      type="button"
      onClick={(evt) => {
        evt.preventDefault();

        changeFavorite(!isFavorite);

        return authorizationStatus === AuthorizationStatus.NO_AUTH
          ? history.push(`/login`)
          : onButtonClick(id, +!isFavorite);
      }}
    >
      <svg className="place-card__bookmark-icon" width="18" height="19">
        <use xlinkHref="#icon-bookmark"></use>
      </svg>
      <span className="visually-hidden">To bookmarks</span>
    </button>
  );
};

FavoriteButton.propTypes = {
  id: PropTypes.number.isRequired,
  favorite: PropTypes.bool.isRequired,
  authorizationStatus: PropTypes.string.isRequired,
  onButtonClick: PropTypes.func.isRequired
};

const mapStateToProps = ({USER}) => ({
  authorizationStatus: USER.authorizationStatus
});

const mapDispatchToProps = (dispatch) => ({
  onButtonClick(id, status) {
    dispatch(setFavorite(id, status));
  }
});

export {FavoriteButton};
export default connect(mapStateToProps, mapDispatchToProps)(FavoriteButton);
