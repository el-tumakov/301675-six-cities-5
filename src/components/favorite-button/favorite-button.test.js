import React from "react";
import renderer from "react-test-renderer";
import {FavoriteButton} from "./favorite-button";
import {noop} from "../../utils";
import {AuthorizationStatus, FavoriteButtonSizes} from "../../const";


jest.mock(`react-router-dom`, () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

it(`Should FavoriteButtonMain render correctly`, () => {
  const tree = renderer
      .create(
          <FavoriteButton
            id={1}
            favorite={true}
            authorizationStatus={AuthorizationStatus.AUTH}
            onButtonClick={noop}
            buttonClassName={`place-card__bookmark`}
            buttonWidth={FavoriteButtonSizes.MAIN_WIDTH}
            buttonHeight={FavoriteButtonSizes.MAIN_HEIGHT}
          />
      ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Should FavoriteButtonRoom render correctly`, () => {
  const tree = renderer
      .create(
          <FavoriteButton
            id={2}
            favorite={false}
            authorizationStatus={AuthorizationStatus.NO_AUTH}
            onButtonClick={noop}
            buttonClassName={`property__bookmark`}
            buttonWidth={FavoriteButtonSizes.ROOM_WIDTH}
            buttonHeight={FavoriteButtonSizes.ROOM_HEIGHT}
          />
      ).toJSON();

  expect(tree).toMatchSnapshot();
});
