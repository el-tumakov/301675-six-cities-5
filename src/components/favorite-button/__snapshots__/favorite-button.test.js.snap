// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoriteButtonMain render correctly 1`] = `
<button
  className="-button--active -button button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height=""
    width=""
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;

exports[`Should FavoriteButtonRoom render correctly 1`] = `
<button
  className=" -button button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height=""
    width=""
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
